src/scripts/main.js

```diff
@@ -1,3 +1,357 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('button');
+const gameScore = document.querySelector('.game-score');
+const cells = document.body.querySelectorAll('.field-cell');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+
+const makeState = (cellsBoard) =>
+  Array.from(cellsBoard)
+    .map((cell, index) =>
+      ({
+        x: index % 4,
+        y: Math.floor(index / 4),
+        value: Number(cell.innerText),
+      }));
+
+let state = makeState(cells);
+
+const clickHandler = (e) => {
+  document.body.addEventListener('keyup', keyUpHandler);
+
+  if (button.innerText === 'Start') {
+    button.innerText = 'Restart';
+    button.className = 'button restart';
+    messageStart.classList.toggle('hidden');
+  } else {
+    gameScore.innerText = 0;
+  }
+  messageWin.className = 'message message-win hidden';
+  messageLose.className = 'message message-lose hidden';
+  resetCells(cells);
+
+  state = makeState(cells);
+  setTwoOrFour(state);
+  setTwoOrFour(state);
+};
+const keyUpHandler = (e) => {
+  switch (e.key) {
+    case 'ArrowLeft':
+      arrowLeft(state);
+
+      if (isWin(state)) {
+        break;
+      }
+      setTwoOrFour(state);
+      isLose(state);
```

Comment:
you can move it outside switch case to not repeat(for every case you're doing the same)

Reply:
I'm rewrite  KeyUpHandler function (switch) as your advice.